src/platforms/esp32/components/avm_builtins/uart_driver.c

```diff
@@ -182,22 +202,28 @@ Context *uart_driver_create_port(GlobalContext *global, term opts)
     term flow_control_term = interop_proplist_get_value_default(opts, FLOW_CONTROL_ATOM, NONE_ATOM);
     term parity_term = interop_proplist_get_value_default(opts, PARITY_ATOM, NONE_ATOM);
 
-    term tx_pin = get_uart_pin_opt(opts, TX_ATOM);
-    term rx_pin = get_uart_pin_opt(opts, RX_ATOM);
-    term rts_pin = get_uart_pin_opt(opts, RTS_ATOM);
-    term cts_pin = get_uart_pin_opt(opts, CTS_ATOM);
+    int tx_pin = get_uart_pin_opt(opts, TX_ATOM);
+    int rx_pin = get_uart_pin_opt(opts, RX_ATOM);
+    int rts_pin = get_uart_pin_opt(opts, RTS_ATOM);
+    int cts_pin = get_uart_pin_opt(opts, CTS_ATOM);
+    if (UNLIKELY(tx_pin < UART_PIN_NO_CHANGE || rx_pin < UART_PIN_NO_CHANGE
```

Comment:
It's better, but code is still inferring that -2 is less than `UART_PIN_NO_CHANGE`, which is a little bit hacky or ugly.

The function should take an additional parameter as suggested (`int *ok`) or we should define a new constant for -2 (`UART_BADARG` ?) and a static assert that it's less than `UART_PIN_NO_CHANGE`.

Reply:
I went with the static assert option, since we only need to check the value of the returned integers. Using the *ok pointer would have required 4 individual checks, rather than one after all pins have been evaluated.